### ISSUE TITLE
__CTOR_END__ has the wrong size

### ISSUE THREAD

**Issue by brooksdavis** [CTSRD-CHERI/lld#17]
The set of .ctor constructors is walked by code like this:
```C
        for (func = &a[0];
            func != &__CTOR_END__;
            func++) {
```
Unfortunately, `__CTOR_LIST__` has no size (presumably because it's not known until link time) and `__CTOR_END__` has a size:
```
000000012001e230  w    O .global_sizes  0000000000000008 .size.__CTOR_END__
```
This means that a comparison of a `__CTOR_LIST__` derived pointer will never equal `__CTOR_END__` with `cnexeq` used for comparison.  It's easy to work around with some `vaddr_t` casts, but we should ensure that `__CTOR_LIST__` and `__CTOR_END__` have a common derivation.

(It's possible this is a clang, but it feels like the real solution will lie in the linker.)

**Comment by arichardson** [comment]
The .size.foo symbols are added by clang so if `.size.__CTOR_LIST__` is missing that seems like it is a problem in clang or the source code. However, this problem should go away once I've finished the capability table implementation in lld.

**Comment by arichardson** [comment]
This is fixed by 630ee65eb676676e0077b8d9d4091776909418a2 and
CTSRD-CHERI/cheribsd@df751bf05dceae3b87db1594646135a2065dcb1f
